src/main/GCNode.cpp

```diff
@@ -332,7 +332,18 @@ void GCNode::Marker::operator()(const GCNode* node)
     // Update mark  Beware ~ promotes to unsigned int.
     node->m_rcmms &= static_cast<unsigned char>(~s_mark_mask);
     node->m_rcmms |= s_mark;
-    ++m_marks_applied;
+    node->visitReferents(this);
+}
+
+void GCNode::CountingMarker::operator()(const GCNode* node)
+{
+    if (node->isMarked()) {
+	return;
+    }
+    // Update mark  Beware ~ promotes to unsigned int.
+    node->m_rcmms &= static_cast<unsigned char>(~s_mark_mask);
```

Comment:
@kmillar sure, I wanted to avoid an extra virtual call but I guess I can call `Marker::operator()(node)` as a non-virtualized call? Not very used to the C++ call mechanics